### PR TITLE
[Table sanity check] sanity check for the table interface_details

### PR DESCRIPTION
**Pull request by akindyakov**
Fixes: #5055

### CODE REVIEW

File: osquery/tests/integration/tables/interface_details.cpp [hunk 1]
```diff
@@ -16,57 +16,54 @@
...
+#endif
+#ifdef OSQUERY_WINDOWS
+      {"friendly_name", NormalType},
+      {"description", NormalType},
+      {"manufacturer", NormalType},
+      {"connection_id", NormalType},
+      {"connection_status", NormalType},
+      {"enabled", Bool},
+      {"physical_adapter", IntType},
+      {"speed", IntType},
```
Comment (guliashvili) [inline osquery/tests/integration/tables/interface_details.cpp]: NonNegativeInt ?

File: osquery/tests/integration/tables/interface_details.cpp [hunk 2]
```diff
@@ -16,57 +16,54 @@
...
+      {"service", NormalType},
+      {"dhcp_enabled", IntType},
```
Comment (guliashvili) [inline osquery/tests/integration/tables/interface_details.cpp]: Bool

File: osquery/tests/integration/tables/interface_details.cpp [hunk 3]
```diff
@@ -16,57 +16,54 @@
...
-  // 4. Perform validation
-  // validate_rows(data, row_map);
+TEST_F(InterfaceDetailsTest, sanity) {
+  QueryData const rows = execute_query("select * from interface_details");
+  auto const row_map = ValidatatioMap{
+      {"interface", NonEmptyString},
+      {"mac", NonEmptyString},
+      {"type", NonNegativeInt},
+      {"mtu", NonNegativeInt},
+      {"metric", IntType},
```
Comment (guliashvili) [inline osquery/tests/integration/tables/interface_details.cpp]: NonNegativeInt

File: osquery/tests/integration/tables/interface_details.cpp [hunk 4]
```diff
@@ -16,57 +16,54 @@
...
+      {"ipackets", NonNegativeInt},
+      {"opackets", NonNegativeInt},
+      {"ibytes", NonNegativeInt},
+      {"obytes", NonNegativeInt},
+      {"ierrors", NonNegativeInt},
+      {"oerrors", NonNegativeInt},
+      {"idrops", NonNegativeInt},
+      {"odrops", NonNegativeInt},
+      {"collisions", NonNegativeInt},
+      {"last_change", IntType},
```
Comment (guliashvili) [inline osquery/tests/integration/tables/interface_details.cpp]: >= -1 ?

File: osquery/tests/integration/tables/interface_details.cpp [hunk 5]
```diff
@@ -16,57 +16,54 @@
...
+      {"description", NormalType},
+      {"manufacturer", NormalType},
+      {"connection_id", NormalType},
+      {"connection_status", NormalType},
+      {"enabled", Bool},
+      {"physical_adapter", IntType},
+      {"speed", IntType},
+      {"service", NormalType},
+      {"dhcp_enabled", IntType},
+      {"dhcp_lease_expires", NormalType},
```
Comment (guliashvili) [inline osquery/tests/integration/tables/interface_details.cpp]: NonNegativeInt ?

File: osquery/tests/integration/tables/interface_details.cpp [hunk 6]
```diff
@@ -16,57 +16,54 @@
...
+      {"dhcp_lease_obtained", NormalType},
```
Comment (guliashvili) [inline osquery/tests/integration/tables/interface_details.cpp]: NonNegativeInt ?

File: osquery/tests/integration/tables/interface_details.cpp (hunk 4 above)
Comment (akindyakov) [reply osquery/tests/integration/tables/interface_details.cpp]: Waiting for your feature #5150 :)

File: osquery/tests/integration/tables/interface_details.cpp (hunk 5 above)
Comment (akindyakov) [reply osquery/tests/integration/tables/interface_details.cpp]: As far as code of receiving this field looks like:
```c++
results[0].GetString("DHCPLeaseExpires", r["dhcp_lease_expires"]);
```
I'm not sure that it is has an integer type.

File: osquery/tests/integration/tables/interface_details.cpp (hunk 6 above)
Comment (akindyakov) [reply osquery/tests/integration/tables/interface_details.cpp]: Unfortunately, same reason as for the `"dhcp_lease_expires"`.

**Comment by akindyakov** [comment]
@guliashvili , why do you think this PR changes API ?

**Comment by akindyakov** [comment]
Oh, yes, you right. Default value was empty string before (for the integer column :) ). Thanks for tagging it!

**Comment by akindyakov** [comment]
I didn't notice you comment about `-1` for the field `link_speed`, sorry about it.
I thought it is signed value, because in the spec file this column it defined as `BIGINT`, which is signed int. Am I wrong?

**Comment by guliashvili** [comment]
I checked in posix implementation, they extract some values using BIGINT_FROM_UINT32, which just casts unsigned to bigint as I understand

**Comment by akindyakov** [comment]
Yes, you are right, for windows the same. Let's use `0` as default value. Thanks!

**Pull request merged**